ipfx/data_set_features.py

```diff
@@ -162,66 +162,97 @@ def extract_cell_features(data_set,
     lu.log_pretty_header("Analyzing cell features:",level=2)
 
     cell_features = {}
+    cell_state = {"failed_fx": False, "fail_fx_message": None}
 
     # long squares
-    lu.log_pretty_header("Long Squares:",level=2)
-    if len(long_square_sweep_numbers) == 0:
-        raise er.FeatureError("No long_square sweeps available for feature extraction")
-
-    lsq_sweeps = data_set.sweep_set(long_square_sweep_numbers)
-    lsq_sweeps.select_epoch("recording")
-    lsq_sweeps.align_to_start_of_epoch("experiment")
-
-    lsq_start, lsq_dur, _, _, _ = stf.get_stim_characteristics(lsq_sweeps.sweeps[0].i, lsq_sweeps.sweeps[0].t)
-
-    lsq_spx, lsq_spfx = extractors_for_sweeps(lsq_sweeps,
-                                              start = lsq_start,
-                                              end = lsq_start+lsq_dur,
-                                              **detection_parameters(data_set.LONG_SQUARE))
-    lsq_an = spa.LongSquareAnalysis(lsq_spx, lsq_spfx, subthresh_min_amp=subthresh_min_amp)
-    lsq_features = lsq_an.analyze(lsq_sweeps)
-    cell_features["long_squares"] = lsq_an.as_dict(lsq_features, [ dict(sweep_number=sn) for sn in long_square_sweep_numbers ])
-
-    if cell_features["long_squares"]["hero_sweep"] is None:
-        raise er.FeatureError("Could not find hero sweep.")
+    try:
+        lu.log_pretty_header("Long Squares:",level=2)
+        if len(long_square_sweep_numbers) == 0:
+            raise er.FeatureError("No long_square sweeps available for feature extraction")
+
+        lsq_sweeps = data_set.sweep_set(long_square_sweep_numbers)
+        lsq_sweeps.select_epoch("recording")
+        lsq_sweeps.align_to_start_of_epoch("experiment")
+        # TODO: fail properly when stim sweeps differ in start/duration
```

Comment:
is this TODO complicated to implement? If so, could you create an issue documenting what needs to happen?

Reply:
yeah this is definitely out of scope for now, and I'm not sure whether it's actually worth the effort of fixing, I've only encountered in one specimen ever I think. I'll check in with ephys folks and create an issue if it seems worth fixing.